src/tools/run-make-support/src/lib.rs

```diff
@@ -244,6 +244,24 @@ pub fn uname() -> String {
     output.stdout_utf8()
 }
 
+/// Search for all files in the current working directory with the extension `ext`,
+/// read their contents and count the
+/// number of regex matches with a given expression (re).
+#[track_caller]
+pub fn count_regex_matches_in_files_with_extension(re: &str, ext: &str) -> usize {
+    let re = regex::Regex::new(re).expect(format!("Regex expression {re} is not valid.").as_str());
+    use io::BufRead;
+    fs_wrapper::read_dir(cwd())
+        .filter_map(|entry| entry.ok())
+        .filter(|entry| entry.path().is_file())
+        .filter(|entry| entry.path().extension().unwrap() == ext)
+        .filter_map(|entry| fs::File::open(&entry.path()).ok())
+        .map(|file| io::BufReader::new(file))
+        .flat_map(|reader| reader.lines().filter_map(|entry| entry.ok()))
+        .filter(|line| re.is_match(line))
+        .count()
+}
```

Comment:
Suggestion: so this is what I had in mind:

```rs
/// Search for all files in the current working directory with the extension `ext`,
/// read their contents and count the
/// number of regex matches with a given expression (re).
#[track_caller]
pub fn count_regex_matches_in_files_with_extension(
    re: &regex::Regex,
    ext: &str,
) -> usize {
    use walkdir::{DirEntry, WalkDir};

    let walker = WalkDir::new(cwd()).into_iter();

    fn is_hidden(entry: &DirEntry) -> bool {
        entry.file_name().to_str().map(|s| s.starts_with(".")).unwrap_or(false)
    }

    let mut count = 0;

    for entry in walker.filter_entry(|e| !is_hidden(e)) {
        let entry = entry.unwrap("failed to get DirEntry");
        if !entry.path().is_file() {
            continue;
        }

        if !entry.path().extension().is_some_and(|e| e == ext) {
            continue;
        }

        let content = fs_wrapper::read(entry.path());
        count += content.lines().filter(|line| re.is_match(line)).count();
    }

    count
}
```

It's a bit different from your version in that it's less compact, but it does not silently swallow errors with `filter_map` + `.ok()` and such. I also opted to pass in a `Regex` instead of the regex string literal, because I think it's more obvious that way (we already re-export the `regex` library).

Let me know what you think.

Reply:
Really neat! I see you handle hidden files and are more methodical with finding the various causes of error. I have placed it in the PR with a few tweaks - for example, I think you meant `expect` here: `entry.unwrap("failed to get DirEntry")`.

Reply:
Yes, it was very illustrative, I didn't actually check if it compiled.